File: backend/src/main/java/csc13001/plantpos/domain/models/Product.java
package csc13001.plantpos.domain.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import csc13001.plantpos.utils.http.JsonModel;
import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@Entity
@Table(name = "products")
@EqualsAndHashCode(callSuper = true)
public class Product extends JsonModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id")
    private Long productId;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @NotBlank(message = "Product name cannot be blank")
    @Column(name = "name")
    private String name = "";

    @Column(name = "description")
    private String description;

    @Column(name = "image")
    private String imageUrl;

    @NotNull(message = "Price cannot be null")
    @Column(name = "price", columnDefinition = "decimal(18,2)")
    private BigDecimal price;

    @PositiveOrZero(message = "Stock must be a non-negative number")
    @Column(name = "stock")
    private int stock;

    @Min(value = 0, message = "Care level must be at least 0")
    @Column(name = "care_level")
    private int careLevel;

    @Column(name = "environment_type")
    private String environmentType = "";

    @Column(name = "size")
    private String size = "";

    @Min(value = 0, message = "Light requirement must be at least 0")
    @Column(name = "light_requirement")
    private int lightRequirement;

    @Min(value = 0, message = "Watering schedule must be at least 0")
    @Column(name = "watering_schedule")
    private int wateringSchedule;

    public Product(
            String name, String description, String imageUrl, BigDecimal price, int stock, int careLevel,
            String environmentType, String size, int lightRequirement, int wateringSchedule, Category category) {
        this.name = name;
        this.description = description;
        this.imageUrl = imageUrl;
        this.price = price;
        this.stock = stock;
        this.careLevel = careLevel;
        this.environmentType = environmentType;
        this.size = size;
        this.lightRequirement = lightRequirement;
        this.wateringSchedule = wateringSchedule;
        this.category = category;
    }
}

File: csc13001-plant-pos-frontend.Core/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace csc13001_plant_pos_frontend.Core.Models;

[Table("products")]
public class Product
{
    [Key]
    [Column("product_id")]
    public int ProductId
    {
        get; set;
    }

    [Column("category_id")]
    [ForeignKey("Category")]
    public int CategoryId
    {
        get; set;
    }
    public Category Category
    {
        get; set;
    }

    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Column("description")]
    public string? Description
    {
        get; set;
    }

    [Column("price", TypeName = "decimal(18,2)")]
    public decimal Price
    {
        get; set;
    }

    [Column("stock")]
    public int Stock
    {
        get; set;
    }

    [Column("care_level")]
    [Range(1, 5)]
    public int CareLevel
    {
        get; set;
    }

    [Column("environment_type")]
    public string EnvironmentType { get; set; } = string.Empty;

    [Column("size")]
    public string Size { get; set; } = string.Empty;

    [Column("light_requirement")]
    public int LightRequirement
    {
        get; set;
    }

    [Column("watering_schedule")]
    public int WateringSchedule
    {
        get; set;
    }
}
